specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json

```diff
@@ -3303,6 +3497,109 @@
+    "ObjectReplicationPolicy": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ObjectReplicationPolicyProperties",
+          "x-ms-client-flatten": true,
+          "description": "Returns the Storage Account Object Replication Policy."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "description": "The replication policy between two storage accounts. Multiple rules can be defined in one policy."
+    },
+    "ObjectReplicationPolicyProperties": {
+      "properties": {
+        "policyId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique id for object replication policy."
+        },
+        "enabledTime": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "Indicates when the policy is enabled on the source account."
+        },
+        "sourceAccount": {
+          "type": "string",
+          "description": "Required. Source account name."
+        },
+        "destinationAccount": {
+          "type": "string",
+          "description": "Required. Destination account name."
+        },
+        "rules": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ObjectReplicationPolicyRule"
+          },
+          "description": "The storage account object replication rules."
+        }
+      },
+      "required": [
+        "sourceAccount",
+        "destinationAccount"
+      ],
+      "description": "The Storage Account ObjectReplicationPolicy properties."
+    },
+    "ObjectReplicationPolicyRule": {
+      "properties": {
+        "ruleId": {
+          "type": "string",
+          "description": "Rule Id is auto-generated for each new rule on destination account. It is required for put policy on source account."
+        },
+        "sourceContainer": {
+          "type": "string",
+          "description": "Required. Source container name."
+        },
+        "destinationContainer": {
+          "type": "string",
+          "description": "Required. Destination container name."
+        },
+        "filters": {
+          "$ref": "#/definitions/ObjectReplicationPolicyFilter",
+          "description": "Optional. An object that defines the filter set."
+        }
+      },
+      "required": [
+        "sourceContainer",
+        "destinationContainer"
+      ],
+      "description": "The replication policy rule between two containers."
+    },
+    "ObjectReplicationPolicyFilter": {
+      "properties": {
+        "prefixMatch": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Optional. Filters the results to replicate only blobs whose names begin with the specified prefix."
+        },
+        "minCreationTime": {
+          "type": "string",
```

Comment:
This should be datetime format. Using string format will make we need to add a lot of code to handle to format convert in Powershell (and other user of SDK), which easy to take bugs.

Reply:
Using datetime format will break the server side. Server side could support datetime format in the future.

Reply:
How would it break server side? It would still be serialized as a string in JSON.

---
In reply to: [401773509](https://github.com/Azure/azure-rest-api-specs/pull/8864#discussion_r401773509) [](ancestors = 401773509)

Reply:
It would also be good to get this fixed up front so we generate SDKs correctly from the beginning.

---
In reply to: [403179698](https://github.com/Azure/azure-rest-api-specs/pull/8864#discussion_r403179698) [](ancestors = 403179698,401773509)